Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Parameters/Wrappers/SpeckleDataObjectWrapper.cs

```diff
@@ -32,14 +49,37 @@ public override required Base Base
   }
 
   /// <summary>
-  /// Converted geometries from DataObject.displayValue.
+  /// Contains a list of <see cref="SpeckleGeometryWrapper"/>.
   /// </summary>
-  public List<GeometryBase> Geometries { get; set; } = [];
+  /// <remarks>
+  /// A list of the wrappers as opposed to the geometry bases allows us to hold on to the color and material information.
+  /// However, this does make syncing of name, props etc. more challenging.
+  /// </remarks>
+  public List<SpeckleGeometryWrapper> Geometries { get; set; } = [];
 
   /// <summary>
-  /// Structured properties from the DataObject.
+  /// Try to keep DataObject.properties as source of truth.
   /// </summary>
-  public SpecklePropertyGroupGoo Properties { get; set; } = new();
+  /// <remarks>
+  /// Property ownership rules:
+  /// - Single geometry → DataObject: Geometry properties become DataObject properties
+  /// - DataObject mutation: DataObject properties overwrite all geometry properties
+  /// - Multiple geometries → DataObject: First geometry's properties become DataObject properties, others inherit
+  /// - Individual geometry mutation: Ignored
+  /// </remarks>
+  public SpecklePropertyGroupGoo Properties
+  {
+    get
+    {
+      SyncGeometriesToDataObject();
```

Comment:
this sync shouldn't be needed in the getter?

Reply:
💯 . they should be side-effect free. updated